src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+    
+    printf("\n*******Enter the x coordinate to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+              "**********************************************\n");
+   scanf("%d",&x);
+   readAndDiscardRestOfLine(stdin);
+   if(x < 0)
+   {
+      printf("\n invalid x coordinate selected \n");
+      goto exit;
+   }
+
+   printf("\n*******Enter the y coordinate to select*******\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   scanf("%d",&y);
+   readAndDiscardRestOfLine(stdin);
+   if(y < 0)
+   {
+      printf("\n invalid y coordinate selected \n");
+      goto exit;
+   }
+
+   printf("\n*******Enter the width to select***************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+            "**********************************************\n");
+   scanf("%d",&width);
+   readAndDiscardRestOfLine(stdin);
+   if(width < 0)
+   {
+      printf("\n invalid width selected \n");
+      goto exit;
+   }
+
+   printf("\n*******Enter the height to select**************\t"
+                     "Acceptable inputs are:\t"
+                     "Min is 0\t"
+                     "Max is based on the current resolution\t"
+             "**********************************************\n");
+   scanf("%d",&height);
+   readAndDiscardRestOfLine(stdin);
+   if(width < 0)
+   {
+      printf("\n invalid height selected \n");
+      goto exit;
+   }
+
+  ret = dsHdmiInScaleVideo(x, y, width, height);
+  if(ret != dsERR_NONE)
+  {
+     UT_LOG_ERROR("Result: Failed to dsHdmiInScaleVideo:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+  } else {
+     UT_LOG_INFO("Result : Passed dsHdmiInScaleVideo paramsx : %d, y : %d, heitht : %d , width : %d",
+                  x, y , width, height);
+  }
+
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select zoom mode.
+*
+* This test function is select zoom mode of selected HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 006@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_zoom_mode(void)
+{
+    gTestID = 6;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select;
+    dsVideoZoom_t mode;
+
+    printf("\n===================Select Zoom Mode================\n");
+    printf("\n**************Availabe Zoom Modes are****************\n");
+    for(dsVideoZoom_t i = dsVIDEO_ZOOM_NONE ; i < dsVIDEO_ZOOM_MAX ; i++)
```

Comment:
?? Technical debt and repeated code everywhere in this file.

- Add the change to a function, so support passing a table and selecting the entry from that.
- And add it to ut-control so that everyone can use it.. 
- if you have common functionliayt across the project, then make it common.

- Create a function that does this and add it to ut-core since everyone will need it?

```c
    for(dsVideoZoom_t i = dsVIDEO_ZOOM_NONE ; i < dsVIDEO_ZOOM_MAX ; i++)
    {
           UT_LOG_INFO("%s : %d", ut_control_GetMapString(dsVideoZoom_mapTable, i) , i);
    }
    printf("\n******************************************************\n");
    scanf("%d",&select);
    readAndDiscardRestOfLine(stdin);
```

Reply:
scanf("%d",&select);
 readAndDiscardRestOfLine(stdin);

added repeated code as function local to the file , since it is not common for all the module .